File: InstagramAPI/Domain/Entities/Notification.cs
namespace Domain.Entities
{
    public class Notification
    {
        public int Id { get; set; }

        public int UserId { get; set; }
        public User User { get; set; }

        public int TriggerById { get; set; }
        public User TriggerBy { get; set; }

        public string Type { get; set; }

        public int? PostId { get; set; }
        public int? StoryId { get; set; }

        public string Message { get; set; }
        public bool IsRead { get; set; }
        public DateTimeOffset CreatedAt { get; set; }
    }
}

File: InstagramUI/types/notification.ts
export interface NotificationType {
  id: number;
  type: string;
  message: string;
  isRead: boolean;
  createdAt: string; 

  triggerById: number;
  triggerByUsername: string;
  triggerByPhoto: string;

  postId?: number; 
  storyId?: number;
}

export interface AppNotification {
  id: number;
  username: string;
  profilePictureUrl: string;
  timestamp: string;
  followedUserId: number | null;
}